sos/report/plugins/coredump.py

```diff
@@ -8,36 +8,101 @@
+    usually under var/lib/systemd/coredump. Symlinks are dropped in the plugin
+    directory to assist in identifying which coredump aligns to which entry in
+    the provided `coredumpctl list` output collection.
+
+    Users may leverage the `executable` option to control which coredumps the
+    plugin will collect. This option takes a case-insensitive python regex
+    string. If provided, only coredump entries for which the EXE field in
+    `coredumpctl list` output matches will be collected.
+
+    The dump files collected are compressed, and users should be aware that
+    when inflated these files can be orders of magnitude larger than their
+    collected sizes.
+    """
+
+    short_desc = 'systemd-coredump related information and dump files'
 
     plugin_name = "coredump"
     profiles = ('system', 'debug')
     packages = ('systemd-udev', 'systemd-coredump')
 
     option_list = [
-        PluginOpt("detailed", default=False,
-                  desc="collect detailed information for every report")
+        PluginOpt("dumps", default=5, desc="number of dump files to collect"),
+        PluginOpt("executable", default='',
+                  desc=("only collect info and dump output for executables "
+                        "matching this regex"))
     ]
 
     def setup(self):
         self.add_copy_spec([
             "/etc/systemd/coredump.conf",
             "/etc/systemd/coredump.conf.d/",
             "/run/systemd/coredump.conf.d/",
-            "/usr/lib/systemd/coredump.conf.d/"
+            "/usr/lib/systemd/coredump.conf.d/",
+            "/usr/lib/systemd/systemd-coredump"
         ])
 
-        self.add_cmd_output("coredumpctl dump")
+        coredump_list = self.collect_cmd_output("coredumpctl list --reverse")
+        if coredump_list['status'] != 0:
+            return
 
-        coredump_list = self.collect_cmd_output("coredumpctl list")
-        if self.get_option("detailed") and coredump_list['status'] == 0:
-            for line in coredump_list["output"].splitlines()[1:]:
-                self.add_cmd_output("coredumpctl info "
-                                    f"{line.split()[4]}")
+        cores_collected = 0
+        for line in coredump_list["output"].splitlines()[1:]:
+            cdump = line.split()
+            pid = cdump[4]
+            exe = cdump[-2]
+            if regex := self.get_option("executable"):
+                if not re.search(regex, exe, re.I):
+                    continue
+            cinfo = self.collect_cmd_output(f"coredumpctl info {pid}")
+            if cinfo['status'] != 0:
+                continue
+            res = cinfo['output']
+            if cores_collected < self.get_option("dumps"):
+                core = re.search(r"(^\s*Storage:(.*)(\(present\)))", res, re.M)
+                try:
+                    core_path = core.groups()[1].strip()
+                    self.add_copy_spec(core_path, tailit=False, sizelimit=100)
```

Comment:
That's a good catch of a current blindspot in `tailit` handling. We don't actually give any indication to the caller that the collection was skipped. Will dig on this.

Reply:
is it worth having an option that allows it to copy a bigger size? I had a coredump that was in 2GB in size recently, and we needed it. I think it would be worthwhile that as an option? thoughts?

Reply:
I'm hesitant to add plugin options around size limiting. Though...what about extending `log_size` to a per-plugin setting similar to `timeout`?